### PR TITLE
Use manifest images to pull specs where possible

### PR DESCRIPTION
**Pull request by nolancon**

### CODE REVIEW

File: pkg/installer/filebuilder.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+	return &fileBuilder{
+		yamlPath:  yamlPath,
+		yamlUrl:   yamlUrl,
+		yamlImage: yamlImage,
+		fileName:  fileName,
+		namespace: namespace,
+	}
+}
+
+// buildInstallerFileSys builds an in-memory filesystem for installer with relevant storageos and
```
Comment (mhmxs) [inline pkg/installer/filebuilder.go:42]: Have you changed those lines above, or just moved it here?

File: pkg/installer/filebuilder.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+		contents, err := fetchImageAndExtractFileFromTarball(location, fb.fileName)
+		if err == nil {
+			return contents, nil
+		}
+		// could not get file from image (may not exist on provided version)
+		// default to release URL
+		location = fb.yamlUrl
+	}
+
+	if util.IsURL(location) {
```
Comment (mhmxs) [inline pkg/installer/filebuilder.go:220]: Maybe an `else if` helps understand the relation between those two.

**Review by mhmxs — APPROVED** [review]

File: pkg/installer/filebuilder.go (hunk 1 above)
Comment (nolancon) [reply pkg/installer/filebuilder.go:42]: `fileBuilder` struct and `newFileBuilder()` are new, to save passing all these parameters around functions.

File: pkg/installer/filebuilder.go (hunk 2 above)
Comment (nolancon) [reply pkg/installer/filebuilder.go]: Just realised an `else if` will mean we bypass the `location = fb.yamlUrl`.

There may be a case where the docker repo exists, but the manifest we want does not (eg `operator-manifests:v2.5.0` will not have `storageos-cluster.yaml`) so then we need to default to the github release url instead and pull from there.

Comment (mhmxs) [reply pkg/installer/filebuilder.go:220]: ohh i see.

**Review by mhmxs — APPROVED** [review]

**Pull request merged**